src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1581,13 +1581,7 @@ protected override void EmitCleanupManaged(ILCodeStream codeStream)
 
     class AnsiStringMarshaller : Marshaller
     {
-        internal override bool CleanupRequired
```

Comment:
tried to mimic stub for `ILCSTRMarshaler` from ilmarshallers.cpp, where its marked with no CleanupRequired.

Reply:
That does not sound right. This needs cleanup to free the buffer.

Reply:
so looking further at `ILCSTRMarshaler` native cleanup is [required](https://github.com/dotnet/runtime/blob/317783916d5695d86b087b45220a28c5d8db9fc6/src/coreclr/vm/ilmarshalers.cpp#L477), but not for managed. Doesnt look like this cg2 implementation makes such a differentiation. Perhaps that is the root of the problem since the failure only occurs during native->Managed string marshalling?

Reply:
I have updated the `CleanupManaged` logic with the one from which seems to work https://github.com/dotnet/runtime/blob/317783916d5695d86b087b45220a28c5d8db9fc6/src/coreclr/vm/ilmarshalers.cpp#L483.